test/integration/endpoints/endpoints_test.go

```diff
@@ -162,6 +166,10 @@ func TestEndpointUpdates(t *testing.T) {
 // does not get endpoints, and after transition to ClusterIP, service gets endpoint,
 // without headless label
 func TestExternalNameToClusterIPTransition(t *testing.T) {
+	_, ctx := ktesting.NewTestContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
A lot of tests calling `kubeapiservertesting.StartTestServerOrDie` need this boiler-plate code although that function internally also does the same thing.

Would it make sense to extend the `kubeapiservertesting.TestServer` struct to include the context, TB and client set interfaces?

Including the client set would allow removing a lot of common code that usually follows, because pretty much every test which creates an API server then also instantiates a client set for it.

The downside is that test code might become a bit weird (?):
```
serverCtx.CoreV1().Pods().Get(serverCtx, ....)
```

Reply:
> Would it make sense to extend the kubeapiservertesting.TestServer struct to include the context, TB and client set interfaces?

`test/utils/ktesting` does that, but let's keep using it separate: https://github.com/kubernetes/kubernetes/pull/124093